src/common/src/Modal/ModalTopBar/ModalTopBar.test.tsx

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
```

Comment:
Looks pretty thorough, nice.
Did you consider testing for `.toBeVisible()` instead of just `.toBeDefined()`? I personally prefer`.toBeInTheDocument()` as it might add some semantic value to the reader over toBeDefined.
Overall it's a bit heavy on the use of data-testid selectors one could argue: https://derekndavis.com/posts/getbytestid-overused-react-testing-library
Thoughts?

Reply:
I agree that it is a bit heavy on the testid attributes, but the use of styled components generates random classes and selectors like `nth-child` or header > div etc will cause a lot of required selector maintenance. Adding an additional class to an element can cause styling collisions (because no shadowroot). Perhaps it takes a bit of patience to accept styled-components, a lot of patience.